pkg/cli/image/manifest/dockercredentials/credential_store_factory.go

```diff
@@ -38,9 +40,48 @@ func (c *credentialStoreFactory) CredentialStoreFor(image string) auth.Credentia
 		return nocreds
 	}
 
-	return dockerregistry.NewStaticCredentialStore(&types.AuthConfig{
+	return NewDynamicCredentialStore(&types.AuthConfig{
```

Comment:
Although it sounds reasonable, this function is not only used by oc-mirror and I'd prefer leaving it as `dockerregistry.NewStaticCredentialStore` because there is no complaints about oc commands not refresh tokens.